app/controllers/subscriber_authentication_controller.rb

```diff
@@ -1,4 +1,6 @@
 class SubscriberAuthenticationController < ApplicationController
+  include SessionsHelper
```

Comment:
Minor aside comment - I'm never too sure whether helpers are meant to be used for sharing controller methods, particularly via the include method which has a different context to views. Your approach seems reasonable here as this seems to be a trodden path in this app, but I do wonder about what the idiomatic Rails way is.

I'm left wondering should helpers that have no view purpose be helpers or should they be concerns.

Rails 5 seems to have added new things to make helper usage in controllers cleaner - which seems to suggest they might be encouraged. There is now a [`helpers`](https://github.com/rails/rails/pull/24866) method which means no helpers need importing and all are available e.g `helpers.authenticate_subscriber`

Anyway aside over 🤔